README.Rmd

```diff
@@ -8,18 +8,120 @@ output: github_document
+
+## Installation
+
 ```{r, eval = FALSE}
+# Obtain the the development version from GitHub:
+# install.packages("devtools")
 devtools::install_github("tidyverse/googledrive")
 ```
 
+## Usage
+
+### Load `googledrive`
+
+```{r}
+library("googledrive")
+```
+
+### Package idiosyncrasies
+
+* All functions begin with the prefix `drive_`
+* Functions and parameters attempt to mimic local file navigating conventions in `R`, such as `list.files`.
+
+### Quick demo
+
+Here's how to list the most recently modified 100 files on your drive. This will kick off your authentication, so you will be sent to your browser to authorize your Google Drive access. The functions here are designed to be pipeable, using `%>%`, however they can also be implemented without.
+
+```{r, eval = FALSE}
+drive_list()
+```
+
+You can narrow the query by specifying a `path` and/or `pattern` you'd like to search within. For example, to search within the folder `foo` for a file named `bar` you could run the following.
+
+```{r, eval = FALSE}
+drive_list(path = "foo", pattern = "bar")
+```
+
+Alternatively, you can pass query parameters to `q` parameter. Accepted search parameters can be found in the [Google Drive API documentation](https://developers.google.com/drive/v3/web/search-parameters). For example, if I wanted to search for all spreadsheets, I could run the following.
+
+```{r, eval = FALSE}
+drive_list(q = "mimeType='application/vnd.google-apps.spreadsheet'")
+```
+
+#### Upload files
+
+We can upload any file type. 
+
+```{r}
+write.table(chickwts, "chickwts.txt")
+drive_chickwts <- drive_upload("chickwts.txt")
+```
+
+We now have a file of class `gfile` that contains information about the uploaded file. 
+
+```{r}
+drive_chickwts
+```
+
+Notice that file was uploaded as a `document`. Since this was a `.txt` document, and we didn't specify the type, `googledrive` assumed it was a document (`?drive_upload` for a full list of assumptions). We can overrule this by using the `type` parameter. Let's delete this file first.
+
+```{r}
+drive_chickwts <- drive_chickwts %>%
+  drive_delete
+```
+
+
+```{r}
+drive_chickwts <- drive_upload("chickwts.txt", type = "spreadsheet")
+```
+
+Let's see if that worked.
+```{r}
+drive_chickwts
+```
+Much better `r emo::ji("tada")`.
+
+#### Share files
+
+Notice the access here says "Shared with specific people". To update the access, we need to change the sharing permissions. Let's say I want anyone with the link to be able to view my dataset.
+
+```{r}
+drive_chickwts <- drive_chickwts %>%
+  drive_share(role = "reader", type = "anyone")
+```
+
+*Notice when I run any of the `googledrive` functions, I am assigning them. This is good practice, since each of these functions is inherintely changing our file on Google Drive, so we want to keep track of that with our `R` object.*
+
```

Comment:
spelling: inherently

I'd reword more along these lines:

> We always assign the return value of googledrive functions back into an R object. This object is of type `gfile`, which holds up-to-date metadata on the associate Drive file. By constantly re-assigning the value, we keep it up-to-date, facilitating all downstream operations.

Tidyverse convention is to surround functions with backticks but not the names of packages or R itself.